onnx/defs/math/defs.cc

```diff
@@ -229,7 +237,14 @@ the tensor elementwise.
     .TypeConstraint(
         "T",
         {"tensor(float16)", "tensor(float)", "tensor(double)"},
-        "Constrain input and output types to float tensors.");
+        "Constrain input and output types to float tensors.")
+    .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+        propagateElemTypeFromInputToOutput(ctx, 0, 0);
+        if (!hasNInputShapes(ctx, 1)) {
+          return;
+        }
+        propagateShapeFromInputToOutput(ctx, 0, 0);
+      });
```

Comment:
It would be useful to define this function (unaryElementWiseOpInferenceFn?) in one place and refer to it in the several functions that can use this.

Reply:
I mean "several ops"

Reply:
You mean extracting this part into one function `propageteShapeTypeFromFirstInput`?

Reply:
Yes